### PR TITLE
bcex: Populate values for amount and price precision.

### PR DESCRIPTION
**Pull request by payala**
Assumed a default of 8 for the amount because I could not find a
better way to determine it and it is needed to have some value in
order to place an order.

For the price, I have parsed the number of decimals used in the last
price string.

### CODE REVIEW

**Comment by payala** [comment]
Sorry, what I said is not correct. It is not necessary to have a value for the amount to be able to place an order.

**Comment by payala** [comment]
@kroitor I'm doing some other changes in the meanwhile, also on bcex, which are not related to the amount and price precision topic, should I make another branch and a different PR?

**Comment by kroitor** [comment]
> should I make another branch and a different PR?

Because this is the same exchange, i think it would be easier to merge all as one, so, you can commit into this branch and I'll merge it soon. But anyways, whatever is more convenient to you.

**Comment by payala** [comment]
Ok, I had to try it, but after repeatedly polling them I just got the following response, which is not surprising to be honest:

> Our tech team replied thatapi_order/ticker  This interface has a limit on the number of returned prices

and later, after insisting that this is not a reply, I also got this:

> Our tech team said  this interface "api_order/ticker"  has all the data you need.

If anyone can distill anything useful from these two sentences, I would be interested. 

Especially, I cannot see a way of obtaining amount precision from anywhere in the API. I considered parsing decimals from the volume, but the volume often times has less decimals than is later allowed when placing an order.

@kroitor I think this is ready to merge.

**Comment by payala** [comment]
After complaining that I thought this was not helpful, they have given a much better response today.

> Our tech team said there are not an api interface which could return the price and precision for all markets but only one market at one time. The documentation has not been added yet. It needs to be obtained based on the actual return results.
>
> Please use api_order/ticker to apply for data in one market, it will return results as following:
image
>
> The results of min_trade, max_trade, number_float and price_float are what you want.

Which is much nicer. Example for BTC/CKUSD:

image

**Comment by payala** [comment]
Well, you know, thank _you!_ for making it so easy to contribute to ccxt.

File: js/bcex.js [hunk 1]
```diff
@@ -297,30 +495,36 @@ module.exports = class bcex extends Exchange {
...
         };
         let response = await this.privatePostApiOrderOrderInfo (this.extend (request, params));
         let order = response['data'];
-        let timestamp = order['created'] * 1000;
+        let timestamp = this.safeInteger (order, 'created') * 1000;
         let status = this.parseOrderStatus (order['status']);
+        let side = this.safeString (order, 'flag');
+        if (side === 'sale')
+            side = 'sell';
+        // Can't use parseOrder because the data format is different btw endpoint for fetchOrder and fetchOrders
```
Comment (kroitor) [inline js/bcex.js]: We design the implementations of parsing functions to handle the differences. Rhe same is true for all parsing methods and all exchanges:

- `parseOrder` will parse the results of `fetchOrder`, `fetchOpenOrders`, `fetchClosedOrders`, `fetchOrders`
- `parseTrade` will parse both public `fetchTrades` and private `fetchMytrades`
- `parseTransaction` will parse `fetchDeposits`, `fetchWithdrawals` and `fetchTransactions`
- `parseTicker` will parse `fetchTicker` and `fetchTickers`
- ...

Thus, parser methods are the actual unification workhorses.

File: js/bcex.js [hunk 2]
```diff
@@ -385,6 +591,11 @@ module.exports = class bcex extends Exchange {
         return this.fetchOrdersByType ('open', symbol, since, limit, params);
     }
 
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        let orders = await this.fetchOrders (symbol, since, limit, params);
+        return this.filterBy (orders, 'status', 'closed');
```
Comment (kroitor) [inline js/bcex.js]: This particular method has a conflict with pagination, the limit of closed orders won't match the limit of all orders, unfortunately. Therefore I switched the value of `has['fetchClosedOrders'] = 'emulated'`. This is explained in detail here: https://github.com/ccxt/ccxt/issues/1896#issuecomment-419090123

**Pull request merged**